### ISSUE TITLE
MPI support on Windows is broken

### ISSUE THREAD

**Issue by Alexander-Barth** [JuliaPackaging/Yggdrasil#8523]
Since HDF5_jll is built with MPI support, NetCDF_jll and the wrapper NCDatasets.jl can use parallel IO.
NetCDF+MPI works well on Linux and MacOS X, however it does not work on Windows.

By looking at the build log files it seems that HDF5_jll (1.14.3) has not detected the MPI libraries on Windows:

https://github.com/JuliaBinaryWrappers/HDF5_jll.jl/releases/download/HDF5-v1.14.3%2B3/HDF5-logs.v1.14.3.x86_64-w64-mingw32-libgfortran3-cxx03-mpi+microsoftmpi.tar.gz

```
Features:
---------
                     Parallel HDF5: no
  Parallel Filtered Dataset Writes: no
                Large Parallel I/O: no
```

Similarily with NetCDF_jll (400.902.211) the parallel support seem to be missing from the Windows NetCDF_jll:

https://github.com/JuliaBinaryWrappers/NetCDF_jll.jl/releases/download/NetCDF-v400.902.211%2B0/NetCDF-logs.v400.902.211.x86_64-w64-mingw32-mpi+microsoftmpi.tar.gz

```
checking whether parallel io is enabled in hdf5... no
checking for library containing H5Dread_chunk... none required
checking for library containing H5Pset_fapl_ros3... none required
checking whether HDF5 allows parallel filters... yes
checking whether szlib was used when building HDF5... yes
checking whether HDF5 library is version 1.10.6 or later... yes
configure: WARNING: Parallel io disabled for netcdf-4 because hdf5 does not support
checking whether parallel I/O is enabled for netcdf-4... no
```

```
# Features
--------
Benchmarks:		no
NetCDF-2 API:		yes
HDF4 Support:		no
HDF5 Support:		yes
NetCDF-4 API:		yes
CDF5 Support:		yes
NC-4 Parallel Support:	no
PnetCDF Support:	no
```

Do you have any ideas what could be the issue? 
CC: @eschnett @visr 
Ref: https://github.com/Alexander-Barth/NCDatasets.jl/issues/122

**Comment by Alexander-Barth** [comment]
Thanks a lot @eschnett ! I don't have access to a Windows system either. Maybe we can leave this issue open in case there would be a volunteer able to debug this issue on Windows?

I will also disable NetCDF with MPI on Windows for now.

(I had checked [build_tarballs.jl](https://github.com/JuliaPackaging/Yggdrasil/blob/868d8ba1f8a5f802567d42df6a37e8fd54a92d35/H/HDF5/build_tarballs.jl), but I did not noticed that the `elif false` despite the comment mentioning that the 64-bit case was disabled. Thank you for pointing that out).

**Comment by visr** [comment]
Yeah disabling MPI support in Windows like HDF5 makes sense for now.

I can do checks on Windows, but right now I don't have the need or time to push for MPI support on Windows.

**Comment by ViralBShah** [comment]
Changed the title to be broadly about MPI on Windows so that this issue is available for others who run into this.

**Comment by amontoison** [comment]
@eschnett 
I was able to fix the issue for 64-bit windows.
I got the same error in MUMPS and using the following link option solved the problem:
`-Wl,--disable-runtime-pseudo-reloc`
--> https://github.com/JuliaPackaging/Yggdrasil/pull/14380
